### ISSUE TITLE
Get Corresponding Token from Output Logits

### ISSUE THREAD

**Issue by sdow1ll** [evolutionaryscale/esm#67]
Hello,

Following #2  #44 and #33 I am trying to use esm3-open-small to pull out logits from ESM3 when given my own sequence. Using `output.per_residue_embedding.shape` I see that the embedding shape is [130, 1536]. How am I able to figure out which token corresponds to which logit? (If I am even getting the logit at all since this seems to be an embedding)

I have done the following so far:

```from esm.models.esm3 import ESM3
from esm.sdk.api import ESMProtein, SamplingConfig
from esm.utils.constants.models import ESM3_OPEN_SMALL


client = ESM3.from_pretrained(ESM3_OPEN_SMALL)

protein = ESMProtein(
    sequence=("QVQLQQSGPGLVKPSQTLSLTCAISGDSVSSYNAVWNWIRQSPSRGLEWLGRTYYRSGWYNDYAESVKSRITINPDTSKNQFSLQLNSVTPEDTAVYYCARSGHITVFGVNVDAFDMWGQGTMVTVSS")
)
protein_tensor = client.encode(protein)

output = client.forward_and_sample(
    protein_tensor, SamplingConfig(return_per_residue_embeddings=True)
)
```

Thank you!

**Comment by ebetica** [comment]
These are embeddings, not logits. I believe you can use the `._forward` function to retrieve logits.

Feel free to reopen this issue if you have further questions.

**Comment by ejmeitz** [comment]
Pretty sure you cannot use `_forward` as to get logits without running into an issue. Running the code below, the error occurs on like 317 inside `forward`, specifically when trying to index `x.shape[1]` on line 313 of esm.py. Looking at the `ProteinTensor` returned by `encode` the tensors passed to `_forward` are just 1D and would cause this to break.

This occurs because inside `forward_and_sample` which is exposed to users, the protein tensor is converted to a `_BatchProteinTensor` and has that extra dimension. If I convert to a batched tensor things work. I wouldn't call this a bug, but its definitely a design choice and as it stands its not really possible to get out logits in a non-hacky way.

```
protein_chain = ProteinChain.from_rcsb("1utn")
protein = ESMProtein.from_protein_chain(protein_chain)
protein_tensor = model.encode(protein)

# DOES NOT WORK
res = model._forward(protein_tensor)

# WORKS
batched_protein = _BatchedESMProteinTensor.from_protein_tensor(protein_tensor)
res = model._forward(batched_protein)
```
